packages/eventindexer/pkg/repo/chart.go

```diff
@@ -38,43 +40,80 @@ func (r *ChartRepository) Find(
 
 	var tx *gorm.DB
 
-	var q string = `SELECT * FROM time_series_data
-	WHERE task = ? AND date BETWEEN ? AND ?
-	ORDER BY date;`
+	var startDate time.Time
+	var endDate time.Time
+	var err error
+	if start != "" {
+		startDate, err = time.Parse("2006-01-02", start)
+		if err != nil {
+			fmt.Print(err)
+			return nil, err
+		}
+	}
+	if end != "" {
+		endDate, err = time.Parse("2006-01-02", end)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-	tx = r.getDB().Raw(q, task, start, end)
+	var q string
+	var args []interface{}
+	if feeTokenAddress == "" && tier == "" {
+		if start != "" && end != "" {
+			q = `SELECT * FROM time_series_data
+			WHERE task = ? AND STR_TO_DATE(date, '%Y-%m-%d') BETWEEN ? AND ?
+			ORDER BY STR_TO_DATE(date, '%Y-%m-%d')`
```

Comment:
Why use ```ORDER BY STR_TO_DATE(date, '%Y-%m-%d')```, rather than ```ORDER BY date```?

Reply:
because in native mysql need to be casted at date data type